### ISSUE TITLE
parse `list { }` as `list{}` 

### ISSUE THREAD

**Issue by bobzhang** [rescript-lang/syntax#690]
For beginners, it is confusing that `list{1,2}`works while `list {1,2}` does not work, it is a whitespace sensitive language now.
I propose that we lex `list\ws+{` as a token, so that we can parse both
cc @IwanKaramazow let me know what you think?

**Comment by butterunderflow** [comment]
After some tries, simply lexing `list\ws+{` as a token will make some regression. Expression like 
```rescript
switch list {
| true => 1
| false => 0
}
```
will not be successfully parsed anymore.

Actually, I think the current behavior of 
```rescript
switch list {
| true => 1
| false => 0
}
```
parsed with no issue but 
```rescript
switch list{
| true => 0
| false => 1
}
```
triggering a syntax error is also not friendly for the beginner.

**Comment by IwanKaramazow** [comment]
Yes, this is not possible. There are too many nuances and error messages that would regress.
